codalab/worker/worker.py

```diff
@@ -191,6 +199,24 @@ def start(self):
                 time.sleep(1 * 60 * 60)
         self.cleanup()
 
+    def terminate_containers(self):
+        """
+        Clean up bundle containers by moving all the existing unfinished bundles to terminal states.
+
+        :returns: the number of unfinished bundles.
+        """
+        unfinished_bundles = []
+        for bundle_uuid in self.runs:
+            run_state = self.runs[bundle_uuid]
+            if run_state.stage != RunStage.FINISHED and run_state.container_id is not None:
+                run_state = run_state._replace(
```

Comment:
Should this be failed?  One could imagine this be restaged so that it gets tried again (somewhat risky if we get into infinite loops).  @nelson-liu might have opinions.

Reply:
Restaging would be amazing, if it's easy to do here.

Reply:
Since this PR already merged, let's add this to another PR. To clarify, do you guys want to move this bundle back to `staged` state, meaning it will be reassigned to a worker (even though it could be the same worker) from **bundle-manager**? Or do you guys just want to rerun it?

Reply:
i was thinking staged...it couldn't be the same worker unless the worker was rerun, right? Since the original worker was terminated.

Reply:
hmm, I was thinking towards the case if the worker terminated earlier got reconnected again...

Reply:
Yes. The moving bundles from workers to `staged` state is something missing and I am working on. Then we can move those terminated bundles to the original `staged` state. Do you guys have other thoughts?